src/cargo/util/lints.rs

```diff
@@ -976,6 +976,20 @@ pub fn unexpected_target_cfgs(
         return Ok(());
     };
 
+    // If we have extra `--check-cfg` args comming from the lints config, we need to
+    // refetch the `--print=check-cfg` with those extra args.
+    let lint_rustflags = pkg.manifest().lint_rustflags();
+    let check_cfg = if lint_rustflags.iter().any(|a| a == "--check-cfg") {
+        Some(target_info.check_cfg_with_extra_args(gctx, &rustc, lint_rustflags)?)
+    } else {
+        None
+    };
+    let check_cfg = check_cfg.as_ref().unwrap_or(&global_check_cfg);
+
+    if !check_cfg.exhaustive {
+        return Ok(());
+    }
```

Comment:
Why was this moved in this commit, rather than always being in this location?

Reply:
Mistake on my side, fixed.